src/pages/storage/forms/UploadVolumeBackupFileForm.tsx

```diff
@@ -0,0 +1,284 @@
+  setUploadState,
+  defaultVolumeName,
+}) => {
+  const { project, isLoading } = useCurrentProject();
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [uploadAbort, setUploadAbort] = useState<AbortController | null>(null);
+  const volumeNameAbort = useState<AbortController | null>(null);
+  const { data: pools = [] } = useStoragePools();
+  const { data: clusterMembers = [] } = useClusterMembers();
+
+  const handleSuccess = (
+    volumeName: string,
+    location: string,
+    pool: string,
+  ) => {
+    const volumeDetailURL = linkForVolumeDetail({
+      location: location,
+      name: volumeName,
+      pool: pool,
+      project: project?.name,
+      type: "custom",
+    } as LxdStorageVolume);
+
+    const message = (
+      <>
+        Created volume{" "}
+        <ResourceLink type="volume" value={volumeName} to={volumeDetailURL} />.
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: async () => navigate(`${volumeDetailURL}/configuration`),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Volume creation failed.", new Error(msg));
+  };
+
+  const handleFinish = () => {
+    queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.volumes;
+      },
+    });
+  };
+
+  const handleUpload = (values: UploadVolumeBackupFileFormValues) => {
+    const uploadController = new AbortController();
+    setUploadAbort(uploadController);
+
+    uploadVolume(
+      values.volumeFile,
+      values.name,
+      project?.name,
+      values.pool,
+      setUploadState,
+      uploadController,
+      values.targetClusterMember,
+    )
+      .then((operation) => {
+        toastNotify.info(
+          <>
+            Upload completed. Now creating volume{" "}
+            <ResourceLabel bold type="volume" value={values.name} />.
+          </>,
+        );
+
+        eventQueue.set(
+          operation.metadata.id,
+          () => {
+            handleSuccess(values.name, values.targetClusterMember, values.pool);
+          },
+          handleFailure,
+          handleFinish,
+        );
+
+        handleCloseModal();
+        navigate(`/ui/project/${project?.name}/storage/volumes`);
+      })
+      .catch((e: AxiosError<LxdSyncResponse<null>>) => {
+        const error = new Error(e.response?.data.error);
+        notify.failure("Volume upload failed", error);
+        formik.setSubmitting(false);
+        setUploadState(null);
+      });
+  };
+
+  const formik = useFormik<UploadVolumeBackupFileFormValues>({
+    initialValues: {
+      name: defaultVolumeName || "",
+      pool: pools?.[0]?.name,
+      volumeFile: null,
+      targetClusterMember: clusterMembers?.[0]?.server_name,
```

Comment:
```suggestion
      targetClusterMember: clusterMembers?.[0]?.server_name ?? "",
```
This needs a fallback, otherwise this gets initialized with null in the non-cluster case and the success notification branch produces an error.